src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -245,16 +247,23 @@ public static async Task<JsonDocument> GetJsonCache(HttpClient client, string ur
                         AllowTrailingCommas = true,
                     });
                     break;
-                default: 
-                    doc = await JsonDocument.ParseAsync(await result.Content.ReadAsStreamAsync());
+                default:
+                    Stream responseStream = await result.Content.ReadAsStreamAsync();
+                    contentLength = responseStream.Length;
+                    doc = await JsonDocument.ParseAsync(responseStream);
                     break;
             }
             
             if (useCache)
             {
                 lock (DataCache)
                 {
-                    MemoryCacheEntryOptions? mce = new() { Size = contentLength };
+                    MemoryCacheEntryOptions? mce = new() 
+                    {
+                        Size = contentLength,
+                        SlidingExpiration = TimeSpan.FromMinutes(30),
+                        AbsoluteExpiration = DateTimeOffset.UtcNow.AddHours(6),
```

Comment:
Nit: there is a built-in way to specify a relative expiration time instead of a strictly absolute expiration time:

```suggestion
                        AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(6),
```

Reply:
Addressed in latest version.